app/models/DraftAvailability.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+sealed trait DraftAvailability
+
+object DraftAvailability {
+
+  def apply[T <: DeparturesSummary](departureSummaries: Option[T]): DraftAvailability = departureSummaries match {
```

Comment:
Is the type parameter needed here? Looks like `DeparturesSummary` is a case class and has no subtypes